workers/datasets_based/src/datasets_based/workers/basic_stats.py

```diff
@@ -0,0 +1,161 @@
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "PreviousStepStatusError", cause, False)
+
+
+class PreviousStepFormatError(BasicStatsWorkerError):
+    """Raised when the content of the previous step has not the expected format."""
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "PreviousStepFormatError", cause, False)
+
+
+class BasicStatsComputationFailed(BasicStatsWorkerError):
+    """Raised when the basic stats computation failed."""
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "BasicStatsComputationFailed", cause, False)
+
+
+def compute_histogram(series = dd.Series) -> Histogram:
+    histogram_range = series.min(), series.max()
+    hist, bin_edges = dd.compute(*da.histogram(series.to_dask_array(), bins=10, range=histogram_range))
+    bin_edges = bin_edges.round(decimals=14)  # round up numpy precision issues
+    return Histogram(hist=hist.tolist(), bin_edges=bin_edges.tolist())
+
+
+def compute_basic_stats_response(dataset: str, config: str, split: str) -> BasicStatsResponse:
+    """
+    Get the response of /basic-stats for one specific dataset on huggingface.co.
+    Args:
+        dataset (`str`):
+            A namespace (user or an organization) and a repo name separated
+            by a `/`.
+        config (`str`):
+            A configuration name.
+        split (`str`):
+            A split name.
+    Returns:
+        `BasicStatsResponse`: An object with the basic_stats.
+    <Tip>
+    Raises the following errors:
+        - [`~workers.sizes.PreviousStepStatusError`]
+          If the the previous step gave an error.
+        - [`~workers.sizes.PreviousStepFormatError`]
+            If the content of the previous step has not the expected format.
+        - [`~workers.sizes.BasicStatsComputationFailed`]
+            If the basic stats computation failed.
+    </Tip>
+    """
+    logging.info(f"get basic-stats for dataset={dataset}")
+
+    try:
+        response = get_response(kind="/parquet-and-dataset-info", dataset=dataset)
+    except DoesNotExist as e:
+        raise DatasetNotFoundError("No response found in previous step for this dataset.", e) from e
+    if response["http_status"] != HTTPStatus.OK:
+        raise PreviousStepStatusError(
+            f"Previous step gave an error: {response['http_status']}. This job should not have been created."
+        )
+    content = response["content"]
+    try:
+        basic_stats: List[BasicColumnStats] = []
+        parquet_files_urls = [
+            parquet_file["url"]
+            for parquet_file in response["content"]["parquet_files"]
+            if parquet_file["config"] == config and parquet_file["split"] == split
+        ]
+        df = dd.read_parquet(parquet_files_urls)
```

Comment:
how would it work for shards? will it read metadata from every shard?

Reply:
It loads all the parquet files only when `compute()` is called (and optimize the processing graph to only use the required shards).

For example, `dd.read_parquet` doesn't raise an error if you pass parquet files with different columns, but `compute()` can.

Reply:
OK, nice!